packages/replay-orchestrator/src/test-run/parallel-tests/merge-test-results.ts

```diff
@@ -135,10 +148,15 @@ const hashScreenshotIdentifier = (
   return stringify(identifier);
 };
 
-export const testRunOutcomeFromDiffResults = (
-  currentResult: TestCaseResult["result"],
-  newScreenshotDiffResults: ScreenshotDiffResult[]
-): TestCaseResult["result"] => {
+export const testRunOutcomeFromDiffResults = ({
```

Comment:
Can this all just be computed from the new state?:

```
const getOverallResult = (screenshotDiffResults) => {
  if (hasNotableDifferences(screenshotDiffResults)) {
     return "fail";
  }
  if (screenshotDiffResults.some(
    (diff) =>
      diff.outcome === "flake"
  )) {
    return "flake"
  }
  return "pass"; 
}
```

Reply:
https://github.com/alwaysmeticulous/meticulous-sdk/pull/355/commits/3d004d297f41df3e595c2393c0fc4e81efd84b36 refactors the logic to only depend of the merged state